source-map.bs

```diff
@@ -274,6 +290,44 @@ fields represent mapped code that also has a mapped name.
 Note: Using file offsets was considered but rejected in favor of using line/column data to avoid becoming
 misaligned with the original due to platform-specific line endings. 
 
+### [=names=] for generated JavaScript code
+
+Source map generators should create a [=mapping=] entry with a [=name=] field for
+the following JavaScript tokens, if
+
+1. The original source language construct maps semantically to the generated
+    JavaScript code.
+
+1. The original source language construct has a name.
+
+Then the [=name=] of the [=mapping=] entry should be the name of the original
+source language construct.
+
+Example: A TypeScript arrow function compiled to a JavaScript function, or a
+local variable that was renamed.
+
+1. The opening parenthesis `(` of [=FunctionDeclaration=], [=FunctionExpression=],
+    [=AsyncFunctionDeclaration=] and [=AsyncFunctionExpression=] for (async) functions.
+
+1. The opening parenthesis `(` or the first letter of the *BindingExpression*
```

Comment:
Wouldn't this conflict with (5), since `x => {}` could potentially require `name` mappings both for the arrow function and for the parameter name. It would almost be better to map the `name` to the `=>` token, since that is unambiguously part of the arrow function itself.

Reply:
Great point. Should we carve out an exception for paren-less, single-argument arrow functions or generally recommend to put the `name` on the `=>` token for all arrow functions?

I'm leaning to the former but an argument could also be made for the latter given that all arrow functions could be handled identically if the `=>` mapping contains the name.